### DISCUSSION TITLE
Schema introspection lib?

### DISCUSSION THREAD

**Discussion by jeandeducla** [diesel-rs/diesel#3269] [Ideas]
I was wondering if this could be possible/a good idea to make `diesel_cli/src/infer_schema_internals` a lib in order to offer the possibility to (for instance) generate custom files from a sql schema?

**Comment by weiznich** [comment]
It is certainly possible to extract this as separate crate. In fact it was a separate (internal) crate before https://github.com/diesel-rs/diesel/pull/1720 landed. The main blocking issues to make it an separate official supported crate again in the future are outlined in this comment: https://github.com/diesel-rs/diesel/pull/1720#issuecomment-391162011

**Reply by jeandeducla** [reply]
Thanks for your answer!

Ok so if I understand your discussion, the main blocking point to making `infer_schema_internals` a separate crate is that the api is not suited as is for external use and it would need to be redesigned from scratch?

Is this feature something that could be/that would make sens to implement in a near future and what would the public api would need to look like?

Thanks again!

**Reply by weiznich** [reply]
The major blocker for me is capacity. We do currently not have the required capacity to design such an API while working on other features that are considered more important by the core team.
It is certainly possible that someone just take the existing code and publish and maintains it as third party crate.
